convert2rhel/actions/system_checks/check_firewalld_availability.py

```diff
@@ -0,0 +1,62 @@
+# Copyright(C) 2023 Red Hat, Inc.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+__metaclass__ = type
+
+import logging
+
+from convert2rhel import actions, systeminfo
+from convert2rhel.systeminfo import system_info
+
+
+logger = logging.getLogger(__name__)
+
+
+class CheckFirewalldAvailability(actions.Action):
+    id = "CHECK_FIREWALLD_AVAILABILITY"
+
+    def run(self):
+        """Error out if the firewalld service is running on the system."""
+        super(CheckFirewalldAvailability, self).run()
+        logger.task("Prepare: Check that firewalld is running")
+
+        if system_info.id == "oracle" and (system_info.version.major == 8 and system_info.version.minor >= 8):
+            if systeminfo.is_systemd_managed_service_running("firewalld"):
+                self.set_result(
+                    level="ERROR",
+                    id="FIREWALLD_RUNNING",
+                    title="Firewalld is running",
+                    description="Firewalld is running and can cause problems during the package replacement phase.",
+                    diagnosis="We've detected that firewalld unit is running and might cause problems related to kernel modules after the conversion is done.",
+                    remediation=(
+                        "Stop firewalld by using the `systemctl stop firewalld` command. This will prevent errors while convert2rhel replaces the system packages"
+                        " and the kernel, whoever, that might not prevent errors from appearing the firewalld logs after the conversion. After the conversion is done,"
+                        " reboot the system to load the new RHEL kernel modules and start firewalld with `systemctl start firewalld` once more."
```

Comment:
Do I recall correctly that stopping the firewall does not help and instead the user needs to change the `CleanupModulesOnExit` firewalld setting or reboot to have the firewall working fine?

I suggest providing the customer with a choice in a pre-conversion report warning message:
- either they manually change the config and then firewalld will work as expected right after convert2rhel finishes,
- or they need to reboot in order to have the firewall working after the conversion

Reply:
@bocekm the first bullet option is correct, the second not so much. They have to stop the firewalld before the conversion, otherwise, the conversion process breaks in the kernel replacement and the system is "dead". So it is required to stop it before. In this case, some rhel kernel modules are loaded into the still-running oracle-kernel. So for the firewall to work correctly again the user has to reboot after the conversion (which he should do anyway). Hopefully that lightness up the situation

Reply:
+1 to @kokesak

Reply:
From @danmyway:
> Rebooting is essential, however you need to get through with the conversion first and that can be done either
1/ stop the firewalld (disable the service to make sure it wont restart during the replacement)
or
2/ change the option CleanupModulesOnExit = yes  to CleanupModulesOnExit = no and keep the firewalld running

Let's add the two options Daniel mentioned to the remediation text so the customer can choose what works better for them.

Reply:
My remediation proposal:
```
"Set the option CleanupModulesOnExit in /etc/firewalld/firewalld.conf to no prior to running convert2rhel:

sed -i -- 's/CleanupModulesOnExit=yes/CleanupModulesOnExit=no/g' /etc/firewalld/firewalld.conf

You can change the option back to yes after the system reboot - that is after the system boots into the RHEL kernel."
```